scripts.Rmd

```diff
@@ -0,0 +1,318 @@
+dependencies, `drake` is able track dependencies and when make.R is rerun,
+only the updated scripts will be rerun.
+
+```{r echo=FALSE,message=FALSE}
+
+# Create Drake Plan
+script_based_plan <- drake_plan(
+  data        = load_data(),
+  munged_data = munge_data(data),
+  hist        = make_histogram(munged_data),
+  fit         = do_regression(munged_data),
+  report      = generate_report(hist, fit)
+)
+
+make(script_based_plan)
+```
+
+```{r}
+
+script_based_config <- drake_config(script_based_plan)
+vis_drake_graph(script_based_config, targets_only = TRUE)
+```
+
+
+Any time a script is updated, either `functions.R` can be sourced again to 
+update the representative function, or the specific `code_to_function()` line 
+can be rerun. `drake` can see that the function is updated and when `make()` 
+is called, it will rerun the updated function and all down-stream functions that
+rely on the output. 
+
+For example, if in "03_histogram.R" were to be updated so the `binwidth` of the 
+histogram was now `0.5` as opposed to the original `0.25`, for `drake` to detect
+that the script has been updated, execute `code_to_function()` on the script.
+
+```{r echo=FALSE}
+writeLines(
+  c("munged_data <- readRDS(\"data/munged_data.RDS\")",
+    "gg <- munged_data %>%",
+    "ggplot() +",
+    "geom_histogram(",
+    "  aes(",
+    "    x = Petal.Width,",
+    "    fill = Species",
+    "    ),",
+    "  binwidth = 0.5) +",
+    "  theme_gray(20)",
+    "ggsave(",
+    "filename = \"data/Petal_Width_vs_Species.PNG\",",
+    "plot = gg",
+    ")",
+    "saveRDS(gg, \"data/Petal_Width_vs_Species.RDS\")"),
+  "03_histogram.R"
+  )
+
+```
+
+```{r}
+# "03_historgram.R" has been updated with a new bin-width of 0.5. Re-running 
+# code_to_function()` to generate the the new function is required for
+# `drake` to recognize that the script has been updated. 
+make_histogram <- code_to_function("03_histogram.R")
+```
+
+`drake` has now marked both `hist` and `report` as outdated, as the function 
+that generates `hist` has been updated, and since `generate_report()` depends on
+`hist`, it also has been outdated.
+
+```{r echo=FALSE,message=FALSE}
+outdated_targets<-outdated(script_based_config)
+#needs to be run, for some reason `drake` is not recognizing the update when compiling
+vis_drake_graph(script_based_config, targets_only = TRUE)
+```
+
+When `make()` is now called, `drake` will rerun only `make_histogram()` to update
+`hist`, and `generate_report()` since it depends on `hist`.
+
+```{r}
+make(script_based_plan)
+```
+
+The dependency graph is now all up to date!
+
+```{r, echo=FALSE}
+vis_drake_graph(script_based_config, targets_only = TRUE)
+```
+
+
+## `drake` and scripts
+
+Hopefully this chapter gave some insight on how to use the `code_to_function()`
+functionality available in `drake`. `code_to_function()`is a quick and dirty way
+to retrofit `drake` to an existing script-based project. _This approach should
```

Comment:
> A large number of preexisting data science workflows consist of numbered, imperative scripts. Many such projects are already too big and cumbersome to seriously refactor. The purpose of `code_to_function()` is to quickly and painlessly allow `drake` to accommodate these situations.

However, for new projects, we strongly recommend that you write functions. Functions help organize your thoughts, and they improve portability, readability, and compatibility with `drake`. For a deeper discussion of functions and their role in `drake`, consider watching the [webinar recording of the 2019-09-23 rOpenSci Community Call](https://ropensci.org/commcalls/2019-09-24).

Even old projects are sometimes pliable enough to refactor into functions, especially with the new [`Rclean`](https://github.com/provtools/rclean) package.